lib/src/hooked_bloc.dart

```diff
@@ -1,8 +1,37 @@
 import 'package:hooked_bloc/hooked_bloc.dart';
 import 'package:hooked_bloc/src/cubit_defaults.dart';
 import 'package:hooked_bloc/src/injection/hook_injection_controller.dart';
+import 'package:provider/provider.dart';
 
+///The entry point for [HookedBloc]
 class HookedBloc {
+
+  /// Initialize [HookedBloc] with default [BlocBase] injection function. This method call is not required. Can be omitted.
+  /// If so, [HookedBloc] will use default [injectionFunction], [builderCondition] and [listenerCondition]
+  ///
+  /// [injectionFunction] used by [useCubit] provides required [BlocBase] class.
+  /// Default implementation tries to find correct [BlocBase] class in widget tree using [Provider.of]
+  /// You can easily hooked up this function with other DI/Service locator libraries like GetIt, like so:
+  /// HookedBloc.initialize(() => getIt);
+  ///
+  /// or setup own provider:
+  /// HookedBloc.initialize(() {
+  ///   return <T extends Object>() {
+  ///     if (T == MyCubit) {
+  ///       return MyCubit() as T;
+  ///     } else {
+  ///       return ...
+  ///     }
+  ///   };
+  /// });
```

Comment:
Make it like 
```
  /// ```dart
  /// HookedBloc.initialize(() {
  ///   return <T extends Object>() {
  ///     if (T == MyCubit) {
  ///       return MyCubit() as T;
  ///     } else {
  ///       return ...
  ///     }
  ///   };
  /// });
  /// ``` 
```
**backtick fences for code blocks**
[source here](https://dart.dev/guides/language/effective-dart/documentation#prefer-backtick-fences-for-code-blocks)

Reply:
good point, done